cedar-policy-validator/src/custom_schema/ast.rs

```diff
@@ -0,0 +1,187 @@
+use cedar_policy_core::{
+    ast::{Id, Name as CName},
+    parser::{Loc, Node},
+};
+use itertools::Either;
+use miette::SourceSpan;
+use nonempty::NonEmpty;
+use smol_str::SmolStr;
+
+use super::to_json_schema::PrimitiveType;
+
+const IPADDR_EXTENSION: &str = "ipaddr";
+const DECIMAL_EXTENSION: &str = "decimal";
+pub(super) const CEDAR_NAMESPACE: &str = "__cedar";
+
+pub type Ident = Node<Id>;
+pub type Str = Node<SmolStr>;
+pub type Name = Either<Ident, Str>;
+pub type Schema = Vec<Node<Namespace>>;
+
+#[derive(Debug, Clone)]
+pub struct Path {
+    pub base: Ident,
+    pub prefix: Vec<Ident>,
```

Comment:
can we reuse Core's `Name` here, or not?

Reply:
No because these fields have `Loc`s attached

Reply:
Got it